ci/generate-workflows.py

```diff
@@ -393,6 +393,8 @@ def asset_patterns(self):
             patterns += ["WezTerm-*.zip"]
         elif ("ubuntu" in self.name) or ("debian" in self.name):
             patterns += ["wezterm-*.deb", "wezterm-*.xz"]
+        elif "alpine" in self.name:
+            patterns += ["/root/packages/wezterm-*.apk", "/root/.abuild/*.rsa.pub"]
```

Comment:
What is the purpose of uploading the `.rsa.pub` here?

Reply:
It's how artefacts are packaged in Alpine CI (`.apk` + public key)
`*.rsa.pub` can be added to `/etc/apk/keys` for the signer to be "trusted" and package can be installed without issues
This can be bypassed with `--allow-untrusted` though

Reply:
For nightly builds, if we regenerate the key on every build, I can see that being annoying for users to keep updating to the new random key.

I think the key should probably be a repo-level secret so that it is consistent across the builds, with the public portion checked in to the repo. We can publish the public key on each release upload so that it is available alongside the `.apk`.

Reply:
I agree but I never dealt with signing/keys in CIs so not sure how to get about that. I assume I could just do below if you set a secret with priv key(?):
```sh
echo "${{ secrets.WEZTERM_SIGN_KEY }}" > /root/.abuild/sign.rsa
```

Reply:
Just be aware that secrets are not available for PRs.

Reply:
I think it will be hard for you to iterate on secrets as you're not the repo owner, so my suggestion to get this through is:

* Do the keygen and upload for tagged releases
* Skip the key upload for nightly builds, and suggest that folks need to use `--allow-untrusted` in any docs that we put together

We can work on adopting secrets as a follow up

Reply:
@wez will it be okay to include key pattern in `upload_asset_tag` function?

Reply:
```py
    def upload_asset_tag(self):
        steps = []

        patterns = self.asset_patterns()
        if "alpine" in self.name:
            patterns.append("$HOME/.abuild/*.rsa.pub")
        glob = " ".join(patterns)
```

Reply:
Yeah, I think that makes sense for `upload_asset_tag`

Reply:
after all I figured out how to add it there
https://github.com/panekj/wezterm/blob/8fb11b6baa2de48550e985720085b09975aca90a/ci/generate-workflows.py#L403-L406